src/Symfony/Bundle/FrameworkBundle/Secrets/SodiumVault.php

```diff
@@ -0,0 +1,165 @@
+            throw new \RuntimeException(sprintf('Unable to create the secrets directory (%s)', $secretsDir));
+        }
+
+        $this->pathPrefix = rtrim($secretsDir, '/'.\DIRECTORY_SEPARATOR).\DIRECTORY_SEPARATOR.basename($secretsDir).'.';
+    }
+
+    public function generateKeys(bool $override = false): bool
+    {
+        if (null === $this->encryptionKey && !empty($this->decryptionKey)) {
+            throw new \LogicException('Cannot generate keys when a decryption key has been provided while instantiating the vault.');
+        }
+
+        if (file_exists($this->pathPrefix.'sodium.decrypt.key')) {
+            $this->loadKeys();
+
+            if (!file_exists($this->pathPrefix.'sodium.encrypt.key')) {
+                $this->export('sodium.encrypt.key', $this->encryptionKey);
+            }
+        }
+
+        if (!$override && null !== $this->encryptionKey) {
+            return false;
+        }
+
+        $this->decryptionKey = sodium_crypto_box_keypair();
+        $this->encryptionKey = sodium_crypto_box_publickey($this->decryptionKey);
+
+        $this->export('sodium.encrypt.key', $this->encryptionKey);
+        $this->export('sodium.decrypt.key', $this->decryptionKey);
+
+        return true;
+    }
+
+    public function seal(string $name, string $value): void
+    {
+        $this->validateName($name);
+        $this->loadKeys();
+        $this->export($name.'.sodium', sodium_crypto_box_seal($value, $this->encryptionKey));
+    }
+
+    public function reveal(string $name): ?string
+    {
+        $this->validateName($name);
+
+        if (!file_exists($file = $this->pathPrefix.$name.'.sodium')) {
+            return null;
+        }
+
+        $this->loadKeys();
+
+        return null !== $this->decryptionKey ? sodium_crypto_box_seal_open(include $file, $this->decryptionKey) : null;
+    }
+
+    public function remove(string $name): bool
+    {
+        $this->validateName($name);
+
+        if (!file_exists($file = $this->pathPrefix.$name.'.sodium')) {
+            return false;
+        }
+
+        return @unlink($file) || !file_exists($file);
+    }
+
+    public function list(bool $reveal = false): array
+    {
+        $secrets = [];
+        $prefix = basename($this->pathPrefix);
+        $prefixLen = \strlen($prefix);
+
+        foreach (scandir(\dirname($this->pathPrefix)) as $name) {
+            if (0 === substr_compare($name, $prefix, 0, $prefixLen) && 0 === substr_compare($name, '.sodium', -\strlen('.sodium'))) {
+                $name = substr($name, $prefixLen, -\strlen('.sodium'));
+                $secrets[$name] = $reveal ? $this->reveal($name) : null;
+            }
+        }
+
+        return $secrets;
+    }
+
+    private function validateName(string $name): void
+    {
+        if (preg_match('/[^-._A-Za-z0-9\x80-\xFF]/', $name)) {
+            throw new \LogicException(sprintf('Secret name is invalid: "%s".', $name));
```

Comment:
`Secret name is invalid: : "%s".`
->
`Secret name ("%s") contains invalid characters. It can only contain letters, digits, underscores, dashes and dots.`